pkg-manager/core/src/install/allProjectsAreUpToDate.ts

```diff
@@ -120,6 +133,31 @@ async function linkedPackagesAreUpToDate (
   )
 }
 
+async function isLocalFileDepUpdated (lockfileDir: string, pkgSnapshot: PackageSnapshot | undefined) {
+  if (!pkgSnapshot) return false
+  const localDepDir = path.join(lockfileDir, (pkgSnapshot.resolution as DirectoryResolution).directory)
+  const manifest = await safeReadPackageJsonFromDir(localDepDir)
+  if (!manifest) return false
+  for (const depField of DEPENDENCIES_OR_PEER_FIELDS) {
+    if (depField === 'devDependencies') continue
+    const manifestDeps = manifest[depField] ?? {}
+    const lockfileDeps = pkgSnapshot[depField] ?? {}
+    for (const depName of Object.keys(manifestDeps)) {
```

Comment:
what if there are more dependencies in the lockfile than there are dependencies in the `package.json`?

Reply:
In this case, some dependencies are removed from the `package.json` file, which I still treat as up-to-date here, so it will not do resolution and changes are not synced to `node_modules/.pnpm`, with the disadvantage that the removed dependencies are still accessible in the `local-file`, I'm not quite sure if there's any other impact.

Reply:
I am not sure about this behaviour. It could be unexpected

Reply:
If you want to leave it as it is, then add a comment about it in the code. So that we'll remember that that is expected behaviour.

Reply:
I will update it to do resolution in this case and add some tests later~

Reply:
Well, even if we do full resolution in this case , reinstalling the dependencies will not remove the dependency(is-psitive) in `node_modules/.pnpm`, maybe it's a problem elsewhere, let's keep this behavior for now.

Reply:
But it will remove the symlink. The package is not removed from `.pnpm` because we have a cache of packages there. It is controlled by this setting: https://pnpm.io/npmrc#modules-cache-max-age

Reply:
Thanks for the explanation, I updated it here.